zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotMetrics.java

```diff
@@ -13,69 +13,126 @@
 import static io.camunda.zeebe.snapshots.impl.SnapshotMetricsDoc.SNAPSHOT_FILE_SIZE;
 import static io.camunda.zeebe.snapshots.impl.SnapshotMetricsDoc.SNAPSHOT_PERSIST_DURATION;
 import static io.camunda.zeebe.snapshots.impl.SnapshotMetricsDoc.SNAPSHOT_SIZE;
+import static io.camunda.zeebe.snapshots.impl.SnapshotMetricsDoc.SNAPSHOT_TRANSFER_DURATION;
 
 import io.camunda.zeebe.util.CloseableSilently;
+import io.camunda.zeebe.util.VisibleForTesting;
+import io.camunda.zeebe.util.collection.MArray;
 import io.camunda.zeebe.util.micrometer.MicrometerUtil;
 import io.micrometer.core.instrument.Clock;
 import io.micrometer.core.instrument.Counter;
 import io.micrometer.core.instrument.DistributionSummary;
 import io.micrometer.core.instrument.Gauge;
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Timer;
+import java.util.List;
 import java.util.concurrent.atomic.AtomicLong;
 
 public final class SnapshotMetrics {
 
-  private final AtomicLong snapshotChunkCount = new AtomicLong();
-  private final AtomicLong snapshotSize = new AtomicLong();
+  private final MArray<AtomicLong> snapshotChunkCountArray = MArray.of(AtomicLong[]::new, 2);
```

Comment:
❓ WTH is Marray.

Reply:
Good question :-D
It's a data structure introduce to limit the performance impact of having metrics with multiple labels. 
As getting the metrics via the `MeterRegistry` is quite slow, they are cached on creation and put into a preallocated array. 
This works where the labels are "enums" which is mostly the case.